api/gateway-operator/v2alpha1/gatewayconfiguration_types.go

```diff
@@ -93,7 +93,7 @@ type GatewayConfigControlPlaneOptions struct {
 // +apireference:kgo:include
 type GatewayConfigDataPlaneOptions struct {
 	// +optional
-	Deployment operatorv1beta1.DataPlaneDeploymentOptions `json:"deployment"`
```

Comment:
Wouldn't it be easier to just redefine `DataPlaneDeploymentOptions` in `v2alpha1` pkg and use that 🤔 ?